Gems/AtomTressFX/Code/Components/HairComponentController.cpp

```diff
@@ -0,0 +1,373 @@
+                Data::AssetBus::MultiHandler::BusDisconnect();
+                if (m_configuration.m_hairAsset.GetId().IsValid())
+                {
+                    Data::AssetBus::MultiHandler::BusConnect(m_configuration.m_hairAsset.GetId());
+                    m_configuration.m_hairAsset.QueueLoad();
+                }
+                else
+                {
+                    RemoveHairObject();
+                }
+            }
+
+            void HairComponentController::OnHairGlobalSettingsChanged(const HairGlobalSettings& hairGlobalSettings)
+            {
+                m_configuration.m_hairGlobalSettings = hairGlobalSettings;
+            }
+
+            void HairComponentController::RemoveHairObject()
+            {
+                if (m_featureProcessor)
+                {
+                    m_featureProcessor->RemoveHairRenderObject(m_renderObject);
+                }
+                m_renderObject.reset();
+            }
+
+            void HairComponentController::OnHairConfigChanged()
+            {
+                // The actual config change to render object happens in the onTick function. We do this to make sure it 
+                // always happen pre-rendering. There is no need to do it before render object created, because the object
+                // will always be created with the updated configuration.
+                if (m_renderObject)
+                {
+                    m_configChanged = true;
+                }
+            }
+
+            void HairComponentController::OnAssetReady(Data::Asset<Data::AssetData> asset)
+            {
+                if (asset.GetId() == m_configuration.m_hairAsset.GetId())
+                {
+                    m_configuration.m_hairAsset = asset;
+                    CreateHairObject();
+                }
+            }
+
+            void HairComponentController::OnAssetReloaded(Data::Asset<Data::AssetData> asset)
+            {
+                OnAssetReady(asset);
+            }
+
+            void HairComponentController::OnActorInstanceCreated([[maybe_unused]]EMotionFX::ActorInstance* actorInstance)
+            {
+                CreateHairObject();
+            }
+
+            void HairComponentController::OnActorInstanceDestroyed([[maybe_unused]]EMotionFX::ActorInstance* actorInstance)
+            {
+                RemoveHairObject();
+            }
+
+            void HairComponentController::OnTick([[maybe_unused]]float deltaTime, [[maybe_unused]]AZ::ScriptTimePoint time)
+            {
+                if (!m_renderObject)
+                {
+                    return;
+                }
+
+                // Config change to renderObject happens on the OnTick, so we know the it occurs before render update.
+                if (m_configChanged)
+                {
+                    m_renderObject->UpdateSimulationParameters(
+                        &m_configuration.m_simulationSettings, 0.035f /*default value in hairRenderObject*/);
```

Comment:
Does this default value ever change? Should it come from somewhere else instead of being a magic number here, to keep things in sync if it ever does change?

Reply:
very good point - this is only done once settings are changed (so basically in editor), but following what I've done in OnTick already I replaced this with proper value and declaration.